webapp/channels/src/components/file_preview_modal/image_preview.tsx

```diff
@@ -27,22 +72,157 @@ export default function ImagePreview({fileInfo, canDownloadFiles}: Props) {
+        };
+
+        document.addEventListener('fullscreenchange', onFullscreenChange);
+
+        return () => {
+            document.removeEventListener('fullscreenchange', onFullscreenChange);
+        };
+    }, [computeScaleOptions]);
+
     if (!canDownloadFiles) {
         return <img src={previewUrl}/>;
     }
 
     return (
-        <a
-            className='image_preview'
-            href='#'
+        <TransformWrapper
+            onTransformed={(e) => setScale(e.state.scale)}
+            ref={transformWrapperRef}
         >
-            <img
-                className='image_preview__image'
-                loading='lazy'
-                data-testid='imagePreview'
-                alt={'preview url image'}
-                src={previewUrl}
-            />
-        </a>
+            {({zoomIn, zoomOut}) => (
+                <div className='image_preview_wrapper'>
+                    <div className='image_preview_zoom_actions__actions'>
+                        <WithTooltip
+                            id='zoomOutTooltip'
+                            title={formatMessage({id: 'file_preview_modal_image_zoom_actions.zoom_out', defaultMessage: 'Zoom out'})}
+                            placement='top'
+                        >
+                            <button
+                                onClick={() => zoomOut()}
+                                className='image_preview_zoom_actions__action-item'
+                                disabled={scale <= 1}
+                            >
+                                <i className='icon icon-minus'/>
+                            </button>
+                        </WithTooltip>
+                        <WithTooltip
+                            id='zoomInTooltip'
+                            title={formatMessage({id: 'file_preview_modal_image_zoom_actions.zoom_in', defaultMessage: 'Zoom in'})}
+                            placement='top'
+                        >
+                            <button
+                                onClick={() => zoomIn()}
+                                className='image_preview_zoom_actions__action-item'
+                            >
+                                <i className='icon icon-plus'/>
+                            </button>
+                        </WithTooltip>
+                        <WithTooltip
+                            id='zoomSelectTooltip'
+                            title={formatMessage({id: 'file_preview_modal_image_zoom_actions.zoom_select', defaultMessage: 'Zoom'})}
+                            placement='top'
+                        >
+                            <select
+                                className='image_preview_zoom_actions__select-item'
+                                value={scale}
+                                onChange={selectZoomLevel}
+                            >
+                                {scaleOptions.map((o) => o.value).includes(scale) ? null : (
```

Comment:
What is this case for? I don't understand the hidden option here

Reply:
So @abhijit-singh had suggested following the design of this zoom library as closely as possible: https://mozilla.github.io/pdf.js/web/viewer.html

In that library when you click on the plus button or otherwise get to a zoom level that isn't selectable from the dropdown, the level of zoom you're on is still shown as the value for the select box, and that's the functionality that this hidden option replicates.

Reply:
Ah, I see. I misread the ternary statement, and I didn't realize it was adding an option when the selected value wasn't in the list at all. This makes sense now